File: Low Level Design (LLD)/Problems/04. AirlineManagementSystem/Java/flight/Aircraft.java
package flight;

import utils.SeatClass;

import java.util.Map;

public class Aircraft {
    private String model;
    private int capacity;
    private Map<SeatClass, Integer> seatConfig;

    public Aircraft(String model, int capacity, Map<SeatClass, Integer> seatConfig) {
        this.model = model;
        this.capacity = capacity;
        this.seatConfig = seatConfig;
    }

    public Map<SeatClass, Integer> getSeatConfig() {
        return seatConfig;
    }
}

File: Low Level Design (LLD)/Problems/04. AirlineManagementSystem/Cpp/flight/src/Aircraft.cpp
#include "../include/Aircraft.h"

Aircraft::Aircraft(const std::string& model, const int& capacity, const std::map<SeatClass, int>& seatConfig) 
    : model(model), capacity(capacity), seatConfig(seatConfig) {}

const std::map<SeatClass, int> Aircraft::getSeatConfig() const {
    return seatConfig;
}